src/protocol/pki.rs

```diff
@@ -0,0 +1,423 @@
+        EnrollmentID(self.0.enrollment_id)
+    }
+
+    fn dump<'py>(&self, py: Python<'py>) -> PyResult<&'py PyBytes> {
+        Ok(PyBytes::new(
+            py,
+            &self
+                .0
+                .clone()
+                .dump()
+                .map_err(|e| ProtocolError::new_err(format!("encoding error: {e}")))?,
+        ))
+    }
+}
+
+gen_rep!(
+    pki_enrollment_info,
+    PkiEnrollmentInfoRep,
+    { .. },
+    [NotFound, reason: Reason],
+);
+
+#[pyclass(extends = PkiEnrollmentInfoRep)]
+#[derive(Clone)]
+pub(crate) struct PkiEnrollmentInfoRepOk;
+
+#[pymethods]
+impl PkiEnrollmentInfoRepOk {
+    #[new]
+    fn new(info_status: PkiEnrollmentInfoStatus) -> (Self, PkiEnrollmentInfoRep) {
+        (
+            Self,
+            PkiEnrollmentInfoRep(pki_enrollment_info::Rep::Ok(info_status.0)),
+        )
+    }
+
+    #[getter]
+    fn enrollment_status(_self: PyRef<'_, Self>) -> PyResult<PkiEnrollmentInfoStatus> {
+        if let pki_enrollment_info::Rep::Ok(infos) = &_self.as_ref().0 {
+            Ok(PkiEnrollmentInfoStatus(infos.clone()))
+        } else {
+            Err(PyAttributeError::new_err("No such field"))
+        }
+    }
+}
+
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct PkiEnrollmentInfoStatus(pki_enrollment_info::PkiEnrollmentInfoStatus);
+
+#[pymethods]
+impl PkiEnrollmentInfoStatus {
+    #[classmethod]
+    #[pyo3(name = "Submitted")]
+    fn submitted(_cls: &PyType, submitted_on: DateTime) -> Self {
+        Self(pki_enrollment_info::PkiEnrollmentInfoStatus::Submitted {
+            submitted_on: submitted_on.0,
+        })
+    }
+
+    #[classmethod]
+    #[pyo3(name = "Accepted")]
+    fn accepted(
+        _cls: &PyType,
+        submitted_on: DateTime,
+        accepted_on: DateTime,
+        accepter_der_x509_certificate: Vec<u8>,
+        accept_payload_signature: Vec<u8>,
+        accept_payload: Vec<u8>,
+    ) -> Self {
+        Self(pki_enrollment_info::PkiEnrollmentInfoStatus::Accepted {
+            submitted_on: submitted_on.0,
+            accepted_on: accepted_on.0,
+            accepter_der_x509_certificate,
+            accept_payload,
+            accept_payload_signature,
+        })
+    }
+
+    #[classmethod]
+    #[pyo3(name = "Rejected")]
+    fn rejected(_cls: &PyType, submitted_on: DateTime, rejected_on: DateTime) -> Self {
+        Self(pki_enrollment_info::PkiEnrollmentInfoStatus::Rejected {
+            submitted_on: submitted_on.0,
+            rejected_on: rejected_on.0,
+        })
+    }
+
+    #[classmethod]
+    #[pyo3(name = "Cancelled")]
+    fn cancelled(_cls: &PyType, submitted_on: DateTime, cancelled_on: DateTime) -> Self {
+        Self(pki_enrollment_info::PkiEnrollmentInfoStatus::Cancelled {
+            submitted_on: submitted_on.0,
+            cancelled_on: cancelled_on.0,
+        })
+    }
+
+    /// This method checks if the status is `Accepted` or not. It is used only in tests
+    fn is_submitted(&self) -> bool {
```

Comment:
Can you use `isinstance` instead ?

Reply:
At the moment status are all wrapped in `PkiEnrollmentStatus`. I made these methods to make testing more convenient. As soon as `anonymous` commands are oxidized we we'll be able to oxidize the `PkiEnrollmentStatus` enum (defined in parsec/backend/protocol/pki.py) too and thus replace these getters by another one which returns a `PkiEnrollmentStatus`.

Using `isinstance` here will require to have a single type for every variants of `PkiEnrollmentStatus`.